### ISSUE TITLE
import/no-duplicates does not catch all duplicates

### ISSUE THREAD

**Issue by tbergq** [benmosher/eslint-plugin-import#1456]
Hi, I came across an example in our codebase where this rule did not catch the duplicate import.

```
import { ProfilePictureCard, SectionCard } from 'our-component-library';
import type { NativeEventHandler } from 'our-component-library';
// Expected ^^ to trigger error, but it did not. 
```
However this is correctly reported: 
```
import { ProfilePictureCard, SectionCard } from 'our-component-library';
import { type NativeEventHandler } from 'our-component-library';
// ^^ Triggers error. 
```

**Comment by ljharb** [comment]
I would not expect `import type` to ever report from this plugin; since that's something flow handles.

**Comment by tbergq** [comment]
Ok, so I might have had wrong expectations then 😞

**Comment by ljharb** [comment]
To be clear, if it's possible and useful to do, I think we should handle it - but i think currently we just utterly ignore any `import type` statements.

**Comment by tbergq** [comment]
Well, I feel the codebase is cleaner if each dependency is only imported once(which is why we use this rule 😊 ).
